src/derivative_utils.jl

```diff
@@ -106,6 +106,17 @@ function calc_J!(nlsolver::NLSolver, integrator, cache::OrdinaryDiffEqMutableCac
   is_compos && (integrator.eigen_est = opnorm(J, Inf))
 end
 
+function calc_J!(nlsolver::NLSolver, integrator, cache::OrdinaryDiffEqConstantCache, is_compos)
```

Comment:
There already exists an implementation for out-of-place function, it's called `calc_J`. So if we need an in-place variant `calc_J!` (which I'm not sure about) we could just define `calc_J!` as `cache.J = calc_J(...)`